src/v/cloud_storage/remote_partition.cc

```diff
@@ -141,7 +141,7 @@ remote_partition::borrow_result_t remote_partition::borrow_next_segment_reader(
             // range.
             auto so = manifest.get_start_kafka_offset().value_or(
               kafka::offset::min());
-            if (config.start_offset < so && config.max_offset > so) {
+            if (config.start_offset() < so && config.max_offset() > so) {
```

Comment:
This works due to implicit conversion to underlying type: `detail::base_named_type::constexpr operator type() const`, which I'm not sure I'm a fan of, but that seems like a big change at this point.

Is it worth putting a `kafka|model::offset_cast` in here to make it more explicit?

Reply:
you are probably right. need to have a pass on this commit

Reply:
did another pass, tried kafka::offset_cast/model::offset_cast when appropriate